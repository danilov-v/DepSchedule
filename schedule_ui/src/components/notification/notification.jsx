import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import { Toast, ToastBody, ToastHeader } from "reactstrap";
import { NotificationManager } from "helpers/notification-manager";

import "./notification.scss";

const DEFAULT_NOTIFICATION_DATA = {
  isOpen: false,
  title: "",
  body: "",
  color: "primary",
};

export function Notification({ data }) {
  const [notificationData, setNotificationData] = useState(data);

  useEffect(() => {
    NotificationManager.subscribe(data => {
      setNotificationData(data);

      setTimeout(() => {
        setNotificationData(DEFAULT_NOTIFICATION_DATA);
      }, 3000);
    });
  }, []);

  return (
    <div className="notification">
      <div className="p-3 my-2 rounded">
        <Toast isOpen={notificationData.isOpen}>
          <ToastHeader icon={notificationData.color}>
            {notificationData.title}
          </ToastHeader>
          <ToastBody>{notificationData.body}</ToastBody>
        </Toast>
      </div>
    </div>
  );
}

Notification.propTypes = {
  data: PropTypes.shape({
    isOpen: PropTypes.bool,
    title: PropTypes.string,
    body: PropTypes.string,
  }),
};

Notification.defaultProps = {
  data: DEFAULT_NOTIFICATION_DATA,
};
