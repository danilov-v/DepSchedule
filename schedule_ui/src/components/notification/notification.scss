.notification {
  position: absolute;
  right: 30px;
  bottom: 30px;
  z-index: 1051;

  .toast {
    background-color: white;
  }
}
