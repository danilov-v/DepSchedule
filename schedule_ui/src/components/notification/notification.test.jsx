import React from "react";
import shallow from "enzyme/shallow";
import { Notification } from "./notification";

const props = {
  data: {
    isOpen: true,
    title: "test",
    body: "some text",
  },
};

describe("notification.jsx", () => {
  describe("<Notification />", () => {
    test("should render and match snapshot", () => {
      const component = shallow(<Notification {...props} />);

      expect(component).toMatchSnapshot();
    });
  });
});
