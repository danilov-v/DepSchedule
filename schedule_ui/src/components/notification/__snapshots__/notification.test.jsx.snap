// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`notification.jsx <Notification /> should render and match snapshot 1`] = `
<div
  className="notification"
>
  <div
    className="p-3 my-2 rounded"
  >
    <Toast
      fade={true}
      isOpen={true}
      tag="div"
      transition={
        Object {
          "appear": true,
          "baseClass": "fade",
          "baseClassActive": "show",
          "enter": true,
          "exit": true,
          "in": true,
          "mountOnEnter": false,
          "onEnter": [Function],
          "onEntered": [Function],
          "onEntering": [Function],
          "onExit": [Function],
          "onExited": [Function],
          "onExiting": [Function],
          "tag": "div",
          "timeout": 150,
          "unmountOnExit": true,
        }
      }
    >
      <ToastHeader
        charCode={215}
        closeAriaLabel="Close"
        tag="strong"
        tagClassName="mr-auto"
        wrapTag="div"
      >
        test
      </ToastHeader>
      <ToastBody
        tag="div"
      >
        some text
      </ToastBody>
    </Toast>
  </div>
</div>
`;
